### PR TITLE
SONARIAC-1022 S6506 Detection should not be thwarted by addition of parameters

### PR DESCRIPTION
**Pull request by petertrr**

### CODE REVIEW

File: iac-extensions/docker/src/test/resources/checks/ClearTextProtocolDowngradeCheck/wget.dockerfile [hunk 1]
```diff
@@ -32,5 +32,22 @@ RUN wget ${OPENSSL3_URL}
...
+# Noncompliant@+1
+RUN wget --secure-protocol=TLSv1_2 https://might-redirect.example.com/install.sh -q -O -
+# Noncompliant@+1
+RUN wget --secure-protocol=TLSv1_2 -q -O - -- https://might-redirect.example.com/install.sh
+
+RUN wget --max-redirect=0 --secure-protocol=TLSv1_2 -q -O - https://might-redirect.example.com/install.sh
+RUN wget --secure-protocol=TLSv1_2 -q -O -  --max-redirect=0 https://might-redirect.example.com/install.sh
+RUN wget --secure-protocol=TLSv1_2 https://might-redirect.example.com/install.sh -q -O - --max-redirect=0
+RUN wget --secure-protocol=TLSv1_2 https://might-redirect.example.com/install.sh --max-redirect=0 -q -O -
+RUN wget --secure-protocol=TLSv1_2 -q -O - --max-redirect=0 -- https://might-redirect.example.com/install.sh
```
Comment (mstachniuk) [inline iac-extensions/docker/src/test/resources/checks/ClearTextProtocolDowngradeCheck/wget.dockerfile:46]: Please add examples for `--max-redirect='0'` and `--max-redirect="0"` - probably a small adjustment in detection logic will be needed.

File: iac-extensions/docker/src/test/resources/checks/ClearTextProtocolDowngradeCheck/curl.dockerfile [hunk 2]
```diff
@@ -32,6 +32,9 @@ RUN curl -foo -L -foo --proto "=foobar" https://redirecttoinsecure.example.com
 # Noncompliant@+1
 RUN curl --proto "=foobar" -foo -L -foo https://redirecttoinsecure.example.com
 
+# Noncompliant@+1
+RUN curl --tlsv1.2 -sSf -L https://might-redirect.example.com/install.sh
```
Comment (mstachniuk) [inline iac-extensions/docker/src/test/resources/checks/ClearTextProtocolDowngradeCheck/curl.dockerfile:36]: Please add examples:
* `RUN curl --tlsv1.2 -sSf --location https://might-redirect.example.com/install.sh` (--location instead -L)
* more examples with `--location` flag
* `RUN curl --tlsv1.2 -sSfL https://might-redirect.example.com/install.sh` (from man page: "Short version options that do not need any additional values can be used immediately next to each other, like for example you can specify all the options -O, -L and -v at once as -OLv.")
* other examples with options containing `L` flag specified at once: `-LOv`, `-OLv`, `-vL` etc.

If it is hard to have support for such flags at once, please create a ticket.

**Review by mstachniuk — COMMENTED** [review]
Good progress, but please see my improvements suggestions

File: iac-extensions/docker/src/test/resources/checks/ClearTextProtocolDowngradeCheck/wget.dockerfile (hunk 1 above)
Comment (petertrr) [reply iac-extensions/docker/src/test/resources/checks/ClearTextProtocolDowngradeCheck/wget.dockerfile:46]: Not sure what would be the proper way to support this (maybe some logic in the CommandDetector? Support for flags with values ignoring quotes on values?), so I added them as additional flag variants.

File: iac-extensions/docker/src/test/resources/checks/ClearTextProtocolDowngradeCheck/curl.dockerfile (hunk 2 above)
Comment (petertrr) [reply iac-extensions/docker/src/test/resources/checks/ClearTextProtocolDowngradeCheck/curl.dockerfile:51]: I managed to add support for grouped flags and even got some new results in the ITs. Maybe this is something that can also be moved to the CommandDetector in the future

Comment (mstachniuk) [reply iac-extensions/docker/src/test/resources/checks/ClearTextProtocolDowngradeCheck/curl.dockerfile:51]: IMO, it makes sense to add support for grouped flags to the command detector as many commands support that. Please create a ticket for that improvement.

**Review by mstachniuk — APPROVED** [review]
LGTM

File: iac-extensions/docker/src/test/resources/checks/ClearTextProtocolDowngradeCheck/curl.dockerfile (hunk 2 above)
Comment (petertrr) [reply iac-extensions/docker/src/test/resources/checks/ClearTextProtocolDowngradeCheck/curl.dockerfile:51]: https://sonarsource.atlassian.net/browse/SONARIAC-1483

**Pull request merged**